pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -97,5 +119,253 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
+	mur := &toolchainv1alpha1.MasterUserRecord{}
+	// We can use the same NamespacedName from the request as the MasterUserRecord will have the same name as the
+	// UserSignup (i.e. the user's username)
+	err = r.client.Get(context.TODO(), request.NamespacedName, mur)
+	if err != nil {
+		// We generally EXPECT the MasterUserRecord to not be found here, so we only deal with other error types
+		if !errors.IsNotFound(err) {
+			// Error reading the MasterUserRecord, requeue the request
+			return reconcile.Result{}, err
+		}
+	} else {
+		// If we successfully found an existing MasterUserRecord then our work here is done, set the status
+		// to Complete and return
+		statusError := r.setStatusComplete(instance, "")
+		if statusError != nil {
+			reqLogger.Error(statusError, "Error updating UserSignup Status to Complete", "Name", instance.Name)
+			return reconcile.Result{}, statusError
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// Check the user approval policy.
+	userApprovalPolicy, err := r.ReadUserApprovalPolicyConfig()
+	if err != nil {
+		statusError := r.setStatusFailedToReadUserApprovalPolicy(instance, err.Error())
+		if statusError != nil {
+			reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+			return reconcile.Result{}, statusError
+		}
+
+		reqLogger.Error(err, "Error reading user approval policy")
+		return reconcile.Result{}, err
+	}
+
+	// If the signup has been explicitly approved (by an admin), or the user approval policy is set to automatic,
+	// then proceed with the signup
+	if instance.Spec.Approved || userApprovalPolicy == config.UserApprovalPolicyAutomatic {
+		if instance.Spec.Approved {
+			statusError := r.setStatusApprovedByAdmin(instance, "")
+			if statusError != nil {
+				reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+				return reconcile.Result{}, statusError
+			}
+		}
+
+		if userApprovalPolicy == config.UserApprovalPolicyAutomatic {
+			statusError := r.setStatusApprovedAutomatically(instance, "")
+			if statusError != nil {
+				reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+				return reconcile.Result{}, statusError
+			}
+		}
+
+		var targetCluster string
+
+		// If a target cluster hasn't been selected, select one from the members
+		if instance.Spec.TargetCluster != "" {
+			targetCluster = instance.Spec.TargetCluster
+		} else {
+			// Automatic cluster selection
+			members := cluster.GetMemberClusters()
+			if len(members) > 0 {
+				targetCluster = members[0].Name
+			} else {
+				statusError := r.setStatusNoClustersAvailable(instance, "No member clusters found")
+				if statusError != nil {
+					reqLogger.Error(statusError, "Error updating UserSignup Status", "UserID", instance.Spec.UserID)
+					return reconcile.Result{}, statusError
+				}
+
+				err = NewSignupError("No target clusters available")
+				reqLogger.Error(err, "No member clusters found")
```

Comment:
Same here. Let's log the original error before setting an error status.

Reply:
Fixed.